transformer_engine/pytorch/te_onnx_extensions.py

```diff
@@ -82,7 +82,11 @@ def compute_in_fp32(g, inp, subgraph, cast_outp):
     If `inp` data type is not FP32, add a cast of `inp` to FP32 and feed that into `subgraph`.
     Then, if `cast_output` is true, cast subgraphs's output back to `inp` data type.
     """
-    inp_dtype = _type_utils.JitScalarType.from_value(inp)
+    inp_dtype = {
+        "Float": _C_onnx.TensorProtoDataType.FLOAT,
+        "Half": _C_onnx.TensorProtoDataType.FLOAT16,
+        "BFloat16": _C_onnx.TensorProtoDataType.BFLOAT16,
+    }[inp.type().scalarType()]
```

Comment:
To be safe we'll want to wrap this in a try/except block because it is possible (due to a "bad" TE subgraph) to get from `inp.type().scalarType()` a string that is not one of these 3. But I can do this later